rotkehlchen/globaldb/cache.py

```diff
@@ -1,71 +1,82 @@
+GENERAL_CACHE_TABLE_NAME = 'general_cache'
+UNIQUE_CACHE_KEYS: set[GeneralCacheType] = {
+    GeneralCacheType.CURVE_POOL_ADDRESS,
+    GeneralCacheType.MAKERDAO_VAULT_ILK,
+    GeneralCacheType.CURVE_GAUGE_ADDRESS,
+}
+
 
-def compute_cache_key(key_parts: Iterable[Union[str, GeneralCacheType]]) -> str:
+def compute_cache_key(key_parts: Iterable[Union[str, GeneralCacheType]]) -> tuple[str, str]:
     """Function to compute cache key before accessing globaldb cache.
     Computes cache key by iterating through `key_parts` and making one string from them.
-    Only tuple with the same values and the same order represents the same key."""
+    Only tuple with the same values and the same order represents the same key. Also returns which
+    cache table the data must me stored in."""
+    unique_cache = list(key_parts)[0] in UNIQUE_CACHE_KEYS
+    cache_table = UNIQUE_CACHE_TABLE_NAME if unique_cache else GENERAL_CACHE_TABLE_NAME
     cache_key = ''
     for part in key_parts:
         if isinstance(part, GeneralCacheType):
             cache_key += part.serialize()
         else:
             cache_key += part
-    return cache_key
+    return cache_key, cache_table
 
 
 # Using any random address here, since length of all addresses is the same
-BASE_POOL_TOKENS_KEY_LENGTH = len(compute_cache_key([GeneralCacheType.CURVE_POOL_TOKENS, ZERO_ADDRESS]))  # noqa: E501
+BASE_POOL_TOKENS_KEY_LENGTH = len(compute_cache_key([GeneralCacheType.CURVE_POOL_TOKENS, ZERO_ADDRESS])[0])  # noqa: E501
 
 
-def globaldb_set_general_cache_values_at_ts(
+def globaldb_set_cache_values_at_ts(
         write_cursor: DBCursor,
         key_parts: Iterable[Union[str, GeneralCacheType]],
         values: Iterable[str],
         timestamp: Timestamp,
 ) -> None:
     """Function to update cache in globaldb. Inserts all values paired with the cache key.
     If any entry exists, overwrites it."""
-    cache_key = compute_cache_key(key_parts)
+    cache_key, cache_table = compute_cache_key(key_parts)
     tuples = [(cache_key, value, timestamp) for value in values]
     write_cursor.executemany(
-        'INSERT OR REPLACE INTO general_cache '
+        f'INSERT OR REPLACE INTO {cache_table} '
         '(key, value, last_queried_ts) VALUES (?, ?, ?)',
         tuples,
     )
 
 
-def globaldb_set_general_cache_values(
+def globaldb_set_cache_values(
         write_cursor: DBCursor,
         key_parts: Iterable[Union[str, GeneralCacheType]],
         values: Iterable[str],
 ) -> None:
     """Function to update cache in globaldb. Inserts all values paired with the cache key.
     If any entry exists, overwrites it. The timestamp is always the current time."""
     timestamp = ts_now()
-    globaldb_set_general_cache_values_at_ts(
+    globaldb_set_cache_values_at_ts(
         write_cursor=write_cursor,
         key_parts=key_parts,
         values=values,
         timestamp=timestamp,
     )
 
 
-def globaldb_get_general_cache_values(
+def globaldb_get_cache_values(
```

Comment:
For unique cache this can return at max 1 value. I would make 2 overloads / split this function in 2 and make the function that is for unique cache return `Optional[str]`. Same for all other getters/setters.

Reply:
This means I will have to rewrite a number of tests written.

should i proceed with it?

Reply:
tests that rely on `get_cache` expect arrays as output so most of them assert agains arrays and length of arrays some of which will have to change if we expect a single string as output as is the case with `unique_cache_keys`

Reply:
I really think that this should be changed in order to be more explicit, but if it is going to be a big change, then let's first see @LefterisJP's opinion on this

Reply:
yeah I would also make this change in a similar vein to what I said above in the set function.

Reply:
you still have not done this, right?

Reply:
Oww sorry I missed it.

So what do you want to be done here exactly if i may ask?

Reply:
What nebolax proposed. If it's too big for you or if it can get complicated no worries, finish all the rest, push it and I can do that myself either in this PR or in a different one.

Reply:
oww okay!

The thing i said was alot of test expecting arrays as output would have to be revamped. But i could do it if he is okay with it

Reply:
Yeah I understand. I like Alexey's suggestion for that. But can't say exactly how much work it would be. If it's just tests it's not really bad and should easily be doable. Just leave it as last and make as separate commit in case it goes south and needs reverting/adjustment.